From 4dcd956d1a79fbb346303e786ac2d9448f191ba3 Mon Sep 17 00:00:00 2001
From: Josh Blum <josh@joshknows.com>
Date: Mon, 3 Jun 2019 23:27:16 -0500
Subject: [PATCH] Remove --no-undefined linker flag for python 3.8

The -Wl,--no-undefined linker flag cannot be consistently applied for
all projects linked against SoapySDR because starting in python3.8 the
--ldflags no longer include the python library, and there are supposed
to be missing symbols which will be provided at runtime by the
interpreter.
---
 Changelog.txt                      | 1 +
 cmake/Modules/SoapySDRConfig.cmake | 4 ----
 2 files changed, 1 insertion(+), 4 deletions(-)

diff --git a/cmake/Modules/SoapySDRConfig.cmake b/cmake/Modules/SoapySDRConfig.cmake
index 95a751c..f74d6c9 100644
--- a/cmake/Modules/SoapySDRConfig.cmake
+++ b/cmake/Modules/SoapySDRConfig.cmake
@@ -74,10 +74,6 @@ if(CMAKE_COMPILER_IS_GNUCXX)
         set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++0x")
     endif()
 
-    #force a compile-time error when symbols are missing
-    set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,--no-undefined")
-    set(CMAKE_MODULE_LINKER_FLAGS "${CMAKE_MODULE_LINKER_FLAGS} -Wl,--no-undefined")
-
     #common warnings to help encourage good coding practices
     add_compile_options(-Wall)
     add_compile_options(-Wextra)
